CHANGELOG.md

```diff
@@ -7,6 +7,10 @@ this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+## [0.4.1] - 2023-10-04
```

Comment:
I don't think it's a good idea to introduce a this big change in a PATCH version change (x.x.1). Those should mainly be used for minor bug fixes. This on the other hand is a change that will require the user to update to pydantic 2.x from 1.x, which I'd consider a breaking change. Considering we're still in pre 1.0.0 versions, I'd say bumping the version to 0.5.0 would be the way to go (unless we want to take the opportunity yo go to 1.0.0, but I think there's still pretty many features I'd want to see before that, but that's just a personal opinion).